### DISCUSSION TITLE
[GTK] Close window action in context menu

### DISCUSSION THREAD

**Discussion by tdgroot** [ghostty-org/ghostty#4198] [Feature Requests, Ideas] [answered]
It would be a nice addition if the context menu for a terminal surface has an action like `Close Window` next to `Split Right/Down` to close the selected split / tab / window / surface.

Right now I have setup a keybinding for this:
```
keybind = ctrl+shift+x=close_window
```

But sometimes I feel like the context menu action is missing when working a bit more with the mouse.

This action also exists in the main menu, so it would also be more consistent

**Comment by mitchellh** [comment]
Can you please further explain your motivation for wanting this? I don't understand and would like to try to.

At the moment, I don't like this idea. There are multiple ways to close the window ("X" button, "Close Window" in the hamburger menu, keybinds). I don't understand why we need to duplicate it again in the context menu.

There is another discussion suggesting adding the full menu to the context menu when the titlebar is hidden, which I'm going to approve. I think that would solve that scenario.

**Reply by tdgroot** [reply]
Hi, you raise some valid points. I think closing a window is very trivial when you just have one window within a tab. However, the only way to close a split is:
- Close the shell within that split (can be challenging sometimes, when SSH connection dies but process hangs)
- Go to the main menu to close that window
- Specify a key binding for closing a window

The third option is what I'm using right now, but sometimes I'm missing the option to just close the split by using the context menu.

I hope this clears up the discussion!

**Reply by mitchellh** [reply]
> (can be challenging sometimes, when SSH connection dies but process hangs)

Unrelated to this, but SSH has an escape for this. Type: `~.` and it'll exit immediately.

I think I'd be more open to having a `close_surface` binding in the context menu. This closes only the nearest item (split, tab, window).

**Reply by tdgroot** [reply]
The SSH control sequences are very useful for this as well, yes, but it doesn't always work reliably when the connection dies.

An action called `close_surface` makes a lot more sense to me as well, but just wanted to clear this up: `close_window` actually closes the currently focused surface.

**Reply by mitchellh** [reply]
> An action called close_surface makes a lot more sense to me as well, but just wanted to clear this up: close_window actually closes the currently focused surface.

They already exist and are separate. `close_window` should be closing the window, all tabs, and all splits.

**Comment by pluiedev** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/1ecf7a23-d124-4ee0-acd0-5a46cf9a0bbd)

Implemented in 1.1 via the "Menu" submenu inside the context menu, which contains all menubar items normally accessible by the hamburger icon.
